### PR TITLE
Ch setup ci #181414554

### PR DESCRIPTION
**Pull request by NNesta**
Coverage Badge
# 1. Name of task
- Update TravisCI to run tests
# 2. Pivot Tracker ID
-[ #181414554](https://www.pivotaltracker.com/story/show/181414554)
# 3. Branch Name
- `ch-setup-ci-#181414554`
# 4. Description of task

This task is about setting CI in the project and adding a coverage badge in the readme. Because Travis currently has some issue we decided to use Github action as it achieves the same result.
# 5. Implementation
This was implemented using the react.js.yml file from the .github/workflows folder where there is yml commands that automate the execution of installation of dependency testing and building and giving the coverage

### CODE REVIEW

File: src/components/app.js [hunk 1]
```diff
@@ -2,9 +2,10 @@ import React from 'react';
 import { useSelector } from 'react-redux';
 
 const App = () =>{ 
-    const barefootColor = useSelector(state => state.dummy.color);
+    // const barefootColor = useSelector(state => state.dummy.color);
```
Comment (rukundo-kevin) [inline src/components/app.js]: Remove this comment if we are not still testing the barefootColor

**Review by rukundo-kevin — CHANGES_REQUESTED** [review]

**Review by rukundo-kevin — DISMISSED** [review]
The ci is running without a problem.

File: .github/workflows/react.js.yml [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+        name: Create the Badge
+        uses: schneegans/dynamic-badges-action@v1.0.0
+        with:
+          auth: ${{ secrets.GIST_SECRET }}
+          gistID: fc852272be18bb21d4a7418ab58e2edc
+          filename: cod-fe__${{ env.BRANCH }}.json
+          label: Test Coverage
+          message: ${{ env.COVERAGE }}
+          color: 'blue,555,daf'
+          namedLogo: mocha
```
Comment (SergeNm) [inline .github/workflows/react.js.yml]: we are not using mocha

File: .github/workflows/react.js.yml [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+          node-version: ${{ matrix.node-version }}
+
+      # basically npm install but only installs from package-lock
+      - run: yarn
+      - run: yarn test
+      - run: yarn build
+
+      # Only run the coverage once
+      - if: ${{ matrix.node-version == '14.x' }}
+        name: Get Coverage for badge
```
Comment (SergeNm) [inline .github/workflows/react.js.yml:34]: Coverage badge?? are you creating coverage baddge?

**Review by SergeNm — DISMISSED** [review]

**Review by iyaremye-faustin — DISMISSED** [review]
Working fine

File: README.md [hunk 4]
```diff
@@ -1,4 +1,6 @@
 <h1>COD-FE</h1>
+<img src="https://github.com/atlp-rwanda/cod-be/actions/workflows/node.js.yml/badge.svg">
+<img src="https://img.shields.io/endpoint?url=https://gist.githubusercontent.com/NNesta/bb0f2667445560c93664f6fc01a3a352/raw/cod-fe__pull_3.json">
```
Comment (SergeNm) [inline README.md]: This coverage will always be for pull request 3?

**Review by SergeNm — DISMISSED** [review]

File: README.md (hunk 4 above)
Comment (NNesta) [reply README.md]: I corrected it

**Review by iyaremye-faustin — APPROVED** [review]
It's ok

**Review by rukundo-kevin — APPROVED** [review]

**Review by SergeNm — APPROVED** [review]

**Review by mucyoalexaxel — APPROVED** [review]
LGTM

**Pull request merged**